### PR TITLE
Add AWS IAM authentication fields in PostgreSQL components

### PR DESCRIPTION
**Pull request by antontroshin**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Issue reference

https://github.com/dapr/components-contrib/pull/3324

### CODE REVIEW

**Review by hhunter-ms — APPROVED** [review]
lgtm

File: daprdocs/content/en/reference/components-reference/supported-bindings/postgresql.md [hunk 1]
```diff
@@ -62,17 +62,21 @@ The AWS authentication token will be dynamically rotated before it's expiration
 
 | Field  | Required | Details | Example |
 |--------|:--------:|---------|---------|
-| `awsRegion` | Y | The AWS Region where the AWS Relational Database Service is deployed to. | `"us-east-1"`
-| `accessKey` | Y | AWS access key associated with an IAM account | `"AKIAIOSFODNN7EXAMPLE"`
-| `secretKey` | Y | The secret key associated with the access key. | `"wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"`
+| `useAWSIAM` | Y | Must be set to `true` to enable the component to retrieve access tokens from AWS IAM. This authentication method only works with AWS Relational Database Service for PostgreSQL databases. | `"true"` |
+| `connectionString` | Y | The connection string for the PostgreSQL database.<br>This must contain the user, which corresponds to the name of the user created inside PostgreSQL that maps to the AWS IAM policy. This connection string should not contain any password. Note that the database name field is denoted by dbname with AWS. | `"host=mydb.postgres.database.aws.com user=myapplication port=5432 dbname=dapr_test sslmode=require"`|
```
Comment (sicoyle) [inline daprdocs/content/en/reference/components-reference/supported-bindings/postgresql.md]: i would add here that the user name is expected to exist.

File: daprdocs/content/en/reference/components-reference/supported-state-stores/setup-postgresql-v1.md [hunk 2]
```diff
@@ -83,6 +83,22 @@ Authenticating with Microsoft Entra ID is supported with Azure Database for Post
 | `azureClientId` | N | Client ID (application ID) | `"c7dd251f-811f-…"` |
 | `azureClientSecret` | N | Client secret (application password) | `"Ecy3X…"` |
 
+### Authenticate using AWS IAM
+
+Authenticating with AWS IAM is supported with all versions of PostgreSQL type components.
```
Comment (sicoyle) [inline daprdocs/content/en/reference/components-reference/supported-state-stores/setup-postgresql-v1.md:88]: pls add that they must exist too. I know it's implied, but better to be explicit in our docs :)

**Review by sicoyle — APPROVED** [review]

**Review by sicoyle — APPROVED** [review]
ty

**Comment by marcduiker** [comment]
@holopin-bot @antontroshin Thanks Anton!

**Pull request merged**